### PR TITLE
Fix pickling WebP

### LINKED ISSUE #4559: Webp images cannot be pickled
### What did you do?

I'm trying to pickle and unpickle webp images. My code works for other image formats, but fails for webp images. For example, run the following code with [this image](https://github.com/python-pillow/Pillow/files/4485578/image.zip) (compressed because GitHub doesn't support webp files):

```python
from PIL import Image
import pickle

image = Image.open('image.webp')
unpickled_image = pickle.loads(pickle.dumps(image))
resized_image = unpickled_image.resize((50, 50))
```

### What did you expect to happen?

A resized `Image` instance should have been created.

### What actually happened?

The following error was thrown:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/seppe/.local/lib/python3.7/site-packages/PIL/Image.py", line 1886, in resize
    self.load()
  File "/home/seppe/.local/lib/python3.7/site-packages/PIL/WebPImagePlugin.py", line 156, in load
    if self.__loaded != self.__logical_frame:
AttributeError: 'WebPImageFile' object has no attribute '_WebPImageFile__loaded'
```

### What are your OS, Python and Pillow versions?

* OS: Ubuntu 19.10
* Python: 3.7.5
* Pillow: 7.1.1

### PR DESCRIPTION
**Pull request by hugovk**
Fixes #4559. 

Changes proposed in this pull request:

 * Fix pickling and unpickling WebP
 * Refactor some similar tests to use `pytest.mark.parametrize` to reduce duplication and make tests independent

### CODE REVIEW

File: src/PIL/WebPImagePlugin.py [hunk 1]
```diff
@@ -38,6 +38,8 @@ class WebPImageFile(ImageFile.ImageFile):
 
     format = "WEBP"
     format_description = "WebP image"
+    __loaded = -1
+    __logical_frame = -1
```
Comment (radarhere) [inline src/PIL/WebPImagePlugin.py:42]:
```suggestion
    __logical_frame = 0
```

I suggest that this be zero. With -1, if I call `tell()` after unpickling,

```python
from PIL import Image
import pickle

image = Image.open('Tests/images/hopper.webp')
unpickled_image = pickle.loads(pickle.dumps(image))
print(unpickled_image.tell())
```

`tell()` reports -1, which we would consider to be invalid.

Comment (hugovk) [reply src/PIL/WebPImagePlugin.py:42]: Good idea. This surfaces another bug:

```
    def test_pickle_image(tmp_path, test_file, test_mode):
        # Act / Assert
        for protocol in range(0, pickle.HIGHEST_PROTOCOL + 1):
>           helper_pickle_string(pickle, protocol, test_file, test_mode)

Tests/test_pickle.py:65:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
Tests/test_pickle.py:40: in helper_pickle_string
    assert im == loaded_im
src/PIL/Image.py:651: in __eq__
    and self.getpalette() == other.getpalette()
src/PIL/Image.py:1324: in getpalette
    self.load()
src/PIL/WebPImagePlugin.py:160: in load
    self._seek(self.__logical_frame)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <PIL.WebPImagePlugin.WebPImageFile image mode=RGB size=128x128 at 0x112A0C130>, frame = 0

    def _seek(self, frame):
>       if self.__physical_frame == frame:
E       AttributeError: 'WebPImageFile' object has no attribute '_WebPImageFile__physical_frame'

src/PIL/WebPImagePlugin.py:150: AttributeError
```


Adding `__physical_frame = 0` shows another:

```
    def test_pickle_image(tmp_path, test_file, test_mode):
        # Act / Assert
        for protocol in range(0, pickle.HIGHEST_PROTOCOL + 1):
>           helper_pickle_string(pickle, protocol, test_file, test_mode)

Tests/test_pickle.py:65:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
Tests/test_pickle.py:40: in helper_pickle_string
    assert im == loaded_im
src/PIL/Image.py:651: in __eq__
    and self.getpalette() == other.getpalette()
src/PIL/Image.py:1324: in getpalette
    self.load()
src/PIL/WebPImagePlugin.py:163: in load
    data, timestamp, duration = self._get_next()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <PIL.WebPImagePlugin.WebPImageFile image mode=RGB size=128x128 at 0x10CF00100>

    def _get_next(self):
        # Get next frame
>       ret = self._decoder.get_next()
E       AttributeError: 'WebPImageFile' object has no attribute '_decoder'

src/PIL/WebPImagePlugin.py:131: AttributeError
```

Will push these to demonstrate on the CI, but do we need to call `_open` somewhere when unpickling?

File: Tests/test_pickle.py [hunk 2]
```diff
@@ -88,3 +82,14 @@ def test_pickle_la_mode_with_palette(tmp_path):
 
         im.mode = "PA"
         assert im == loaded_im
+
+
+def test_pickle_tell():
```
Comment (hugovk) [inline Tests/test_pickle.py:87]: Thanks!

I should have also added this:

```suggestion
@skip_unless_feature("webp")
def test_pickle_tell():
```

Comment (radarhere) [reply Tests/test_pickle.py:87]: Yep, I've pushed that to my PR

**Pull request merged**